pkgs/development/tools/build-managers/goredo/fix-tests.diff

```diff
@@ -0,0 +1,36 @@
+diff -ur goredo-1.4.1/t/apenwarr/105-sympath/all.do goredo-1.4.1.new/t/apenwarr/105-sympath/all.do
+--- goredo-1.4.1/t/apenwarr/105-sympath/all.do	2021-04-29 14:03:11.000000000 +0200
++++ goredo-1.4.1.new/t/apenwarr/105-sympath/all.do	2021-05-10 15:04:46.912799930 +0200
+@@ -11,12 +11,13 @@
+ 	(
+ 		cd y/x/x/x/x/x
+ 		IFS=$(printf '\n')
++		_wd=$(env pwd)
+ 		redo-ifchange static x/x/x/static $PWD/static \
+-			$(/bin/pwd)/static /etc/passwd
++			$_wd/static /etc/passwd
+ 		# goredo: that symlink path is not resolving even at OS level
+ 		# redo-ifchange $PWD/../static 2>/dev/null && exit 35
+ 		redo-ifchange 1.dyn x/x/x/2.dyn $PWD/3.dyn \
+-			 $PWD/../4.dyn $(/bin/pwd)/5.dyn
++			 $PWD/../4.dyn $_wd/5.dyn
+ 	)
+ 	[ -e y/1.dyn ] || exit $((iter + 1))
+ 	[ -e y/2.dyn ] || exit $((iter + 2))
+diff -ur goredo-1.4.1/t/apenwarr/clean.do goredo-1.4.1.new/t/apenwarr/clean.do
+--- goredo-1.4.1/t/apenwarr/clean.do	2021-04-29 14:03:11.000000000 +0200
++++ goredo-1.4.1.new/t/apenwarr/clean.do	2021-05-10 15:06:07.099591609 +0200
+@@ -1,3 +1,3 @@
+-/bin/ls [0-9s][0-9][0-9]*/clean.do |
++env ls [0-9s][0-9][0-9]*/clean.do |
+ sed 's/\.do$//' |
+ xargs redo
+diff -ur goredo-1.4.1/t/redo-sh.tests/clean.do goredo-1.4.1.new/t/redo-sh.tests/clean.do
+--- goredo-1.4.1/t/redo-sh.tests/clean.do	2021-04-29 14:03:11.000000000 +0200
++++ goredo-1.4.1.new/t/redo-sh.tests/clean.do	2021-05-10 15:02:41.607562802 +0200
+@@ -1,4 +1,4 @@
+ for f in * ; do
+     [ -d $f ] || continue
+-    find $f ! -name test -delete
++    find $f ! -name test -delete || true
```

Comment:
If this line fails all the time we can also remove it altogether.

Reply:
It does always fail during a single build, but removing the line would impede testing during incremental development.

Reply:
Do you have the sandbox deactivated? It shouldn't matter if you build it the first or 100th try.

Reply:
That was not what I meant. When you build it with nix-build then, yes, it fails without the patch, every time. When I do interactive development with nix-shell, though, I may run the tests multiple times, in which case I'd like to keep this line to get the old test results cleaned up.